### PR TITLE
added home page with categories and subcategories

### PR DESCRIPTION
**Pull request by ReinisVelmeris**

### CODE REVIEW

**Comment by Jekabs-Leinis** [comment]
image

Resolve conflicts please

File: src/common/models/advertisementModels.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export class Advertisement {
+  constructor(props = {}) {
+      Object.assign(this, props);
+  }
+
+  id!: number;
```
Comment (Jekabs-Leinis) [inline src/common/models/advertisementModels.ts]: Will be `string` in firebase.

Please also change in other models

File: src/views/HomeView.vue [hunk 2]
```diff
@@ -1,9 +1,158 @@
...
+import { useFirestore } from 'vuefire'
+import { collection, getDocs } from 'firebase/firestore'
+
+// FETCH ALL CATEGORIES AND SUBCATEGORIES
+const categories = ref<Array<{ name: string, iconURL: string, subCategories: Array<{ name: string, properties: [] }> }>>([]);
+
+const db = useFirestore()
+getDocs(collection(db, 'categories')).then((docs) => {
+  docs.forEach((doc) => {
+    console.log(doc.data());
```
Comment (Jekabs-Leinis) [inline src/views/HomeView.vue]: Leftover from debugging?

File: src/common/stores/advertisementStore.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { ref, computed } from 'vue'
+import { defineStore } from 'pinia'
+import { User } from '../models/advertisementModels'
+
+export const useUserStore = defineStore('user', () => {
+  const currentAdvertisement = ref({})
+  const isLoggedIn = computed(() => currentAdvertisement.value instanceof Advertisement)
```
Comment (Jekabs-Leinis) [inline src/common/stores/advertisementStore.ts]: what does this even mean in the context of an advertisement?

File: src/common/stores/advertisementStore.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { ref, computed } from 'vue'
+import { defineStore } from 'pinia'
+import { User } from '../models/advertisementModels'
+
+export const useUserStore = defineStore('user', () => {
```
Comment (Jekabs-Leinis) [inline src/common/stores/advertisementStore.ts]: Why is this even needed? Is it even used anywhere?

File: src/views/HomeView.vue [hunk 5]
```diff
@@ -1,9 +1,158 @@
 <script setup lang="ts">
+import { ref, computed } from 'vue'
+import { useFirestore } from 'vuefire'
+import { collection, getDocs } from 'firebase/firestore'
+
+// FETCH ALL CATEGORIES AND SUBCATEGORIES
+const categories = ref<Array<{ name: string, iconURL: string, subCategories: Array<{ name: string, properties: [] }> }>>([]);
```
Comment (Jekabs-Leinis) [inline src/views/HomeView.vue]: Would `ref<Category[]>([])` work? Why create the class if you're not gonna use it

File: src/views/HomeView.vue [hunk 6]
```diff
@@ -1,9 +1,158 @@
...
+
+const db = useFirestore()
+getDocs(collection(db, 'categories')).then((docs) => {
+  docs.forEach((doc) => {
+    console.log(doc.data());
+    categories.value.push({
+      name: doc.data().name,
+      iconURL: doc.data().icon_url,
```
Comment (Jekabs-Leinis) [inline src/views/HomeView.vue]: I had the same need to map between snake and camel cases with users and I solved it by changing the DB schema and logic to store the data in camelCase.

Snake_case is the accepted format for SQL databases, but camelCase will work just fine for nosql json docs.

File: src/views/HomeView.vue [hunk 7]
```diff
@@ -1,9 +1,158 @@
...
+
+function numberOfRows() {
+  let total = categories.value.length;
+  let rows = 0;
+  while (total > 0) {
+    rows++;
+    total -= rows + 1; // because we start with 2 columns in the first row
+  }
+  return rows;
+}
```
Comment (Jekabs-Leinis) [inline src/views/HomeView.vue]: A `return Math.floor(Math.floor(Math.sqrt(1 + 8 * total) - 1) / 2);` would be slightly more optimal, but a tad less understandable, so it's up to you if you want to switch

File: src/views/HomeView.vue [hunk 8]
```diff
@@ -1,9 +1,158 @@
...
-  <main>
+  <div class="d-flex justify-content-center">
+    <div class="container-lg m shadow p-5">
+      <h1>SSLV</h1>
+      <p>Tavs ceļš uz lieliskiem ātrajdiem un grandiozām pārdošanām!</p>
 
-  </main>
+      <div class="row d-flex">
+        <div class="col-md-4 d-flex flex-column" v-for="category in categories" :key="category.name">
+          <div class="card mb-3 flex-grow-1 position-relative"> <!-- Add position-relative here -->
```
Comment (Jekabs-Leinis) [inline src/views/HomeView.vue]: Remove html comments here and elsewhere

File: src/views/HomeView.vue [hunk 9]
```diff
@@ -1,9 +1,158 @@
...
+            <img :src="`../common/assets/icons/${category.iconURL}`" alt="icon" class="card-img-top mx-auto mt-2">
+            <div class="card-body">
+              <h5 class="card-title">{{ category.name }}</h5>
+              <a href="#" class="card-link">Action</a>
+              <a href="#" class="card-link">Another action</a>
```
Comment (Jekabs-Leinis) [inline src/views/HomeView.vue]: Nav jau nesmuki, bet uzspiest uz šiem iespējams nebūs

File: src/views/HomeView.vue [hunk 10]
```diff
@@ -1,9 +1,158 @@
...
+  return rows;
+}
+
+function getColumnClass(row: number) {
+  return `col-${12 / (row + 2)}`;
+}
+function getCategory(row: number, col: number) {
+  const index = (row * (row + 1)) / 2 + col;
+  return categories.value[index];
+}
```
Comment (Jekabs-Leinis) [inline src/views/HomeView.vue]: Unused?

**Review by Jekabs-Leinis — CHANGES_REQUESTED** [review]

File: src/views/HomeView.vue (hunk 10 above)
Comment (ReinisVelmeris) [reply src/views/HomeView.vue]: removed

File: src/views/HomeView.vue (hunk 8 above)
Comment (ReinisVelmeris) [reply src/views/HomeView.vue]: removed

File: src/views/HomeView.vue (hunk 7 above)
Comment (ReinisVelmeris) [reply src/views/HomeView.vue]: removed

File: src/views/HomeView.vue (hunk 6 above)
Comment (ReinisVelmeris) [reply src/views/HomeView.vue]: removed console.log, added camelCase in schema

File: src/views/HomeView.vue (hunk 5 above)
Comment (ReinisVelmeris) [reply src/views/HomeView.vue]: Went by the user example, changed this line to include the category class

File: src/common/stores/advertisementStore.ts (hunk 4 above)
Comment (ReinisVelmeris) [reply src/common/stores/advertisementStore.ts]: nope, removed

File: src/common/stores/advertisementStore.ts (hunk 3 above)
Comment (ReinisVelmeris) [reply src/common/stores/advertisementStore.ts]: nothing, started working on advertisements but did not finish

File: src/views/HomeView.vue (hunk 2 above)
Comment (ReinisVelmeris) [reply src/views/HomeView.vue]: yep

File: src/common/models/advertisementModels.ts (hunk 1 above)
Comment (ReinisVelmeris) [reply src/common/models/advertisementModels.ts]: changed

File: src/views/HomeView.vue (hunk 9 above)
Comment (ReinisVelmeris) [reply src/views/HomeView.vue]: tos es noņemšu, bija doma varbūt izmantot, bet nevaru izdomāt ko tur ielikt

File: src/views/HomeView.vue (hunk 5 above)
Comment (Jekabs-Leinis) [reply src/views/HomeView.vue]: If you check out latest master I have significantly changes the User approach, to what I think is better

**Review by Jekabs-Leinis — APPROVED** [review]

**Review by Jekabs-Leinis — APPROVED** [review]

**Pull request merged**